src/AutorestSwift/View Models/ResponseViewModel.swift

```diff
@@ -53,20 +56,33 @@ struct ResponseViewModel {
 
         // check if the request body schema type is object, store the object type of the response body
         let schemaResponse = response as? SchemaResponse
-        self.objectType = schemaResponse?.schema.swiftType(optional: false)
 
+        let strategy: ResponseBodyType
         if let pagingMetadata = operation.extensions?["x-ms-pageable"]?.value as? [String: String],
             let pagingNames = Language.PagingNames(from: pagingMetadata) {
             let arrayElements = (schemaResponse?.schema.properties ?? []).compactMap { $0.schema as? ArraySchema }
             guard arrayElements.count == 1
             else { fatalError("Did not find exactly one array type for paged collection.") }
-            self.strategy = .pagedBody
+            strategy = .pagedBody
             self.pagingNames = pagingNames
             self.pagedElementClassName = arrayElements.first?.elementType.name
+
+            if let elementType = pagedElementClassName, self.pagingNames != nil {
+                self.objectType = "PagedCollection<\(elementType)>"
+            } else {
+                self.objectType = schemaResponse?.schema.swiftType(optional: false) ?? "Void"
+            }
         } else {
-            self.strategy = objectType != nil ? .body : .noBody
             self.pagingNames = nil
             self.pagedElementClassName = nil
+            if let objectType = schemaResponse?.schema.swiftType(optional: false) {
+                strategy = .body
+                self.objectType = objectType
+            } else {
+                strategy = .noBody
+                self.objectType = "Void"
+            }
         }
+        self.strategy = strategy.rawValue
```

Comment:
I don't see any logic setting `intBody`, `stringBody` or `jsonBody`.

Reply:
it is set in ./View Models/ExceptionResponseViewModel.swift